src/config/files.c

```diff
@@ -155,15 +155,6 @@ files_get_config_path(const char* const config_base)
     return result;
 }
 
-/**
- * Get the full path by appending the given location to the project base path.
- *
- * @param location The location (directory or file) to append to the project base path.
- * @return The full path obtained by appending the location to the project base path.
- *
- * @note The returned value must be freed using g_free when it is no longer needed
- *       to prevent memory leaks.
- */
```

Comment:
Comment about function should be in the .c file and not the header.
The dev is reading the .c file and shouldnt have to switch to the header just to read the comment.
Maybe some IDEs take it from the .h file but I'm not a fan of this.
True for all other cases in this commit as well.

And please break up this commit. The docu change should be a commit of its own. And not mixed with the char one.